measures/ApplyUpgrade/measure.rb

```diff
@@ -506,6 +621,70 @@ def run(model, runner, user_arguments)
       end
     end # end hpxml.buildings.each_with_index do |hpxml_bldg, unit_number|
 
+    # Maintain Duct System Curve
+    if use_autosizing_limits_and_maintain_duct_system_curve == 'true'
+      if File.exist?(hpxml_path)
+        hpxml = HPXML.new(hpxml_path: hpxml_path)
+      else
+        runner.registerWarning("ApplyUpgrade measure could not find '#{hpxml_path}'.")
+        return true
+      end
+
+      hpxml.buildings.each_with_index do |hpxml_bldg, _unit_number|
+        air_distribution_airflows = get_air_distribution_airflows(hpxml_bldg)
+
+        if !baseline_max_airflow_cfm.nil? && !air_distribution_airflows.empty? # ducted -> ducted
+          upgrade_max_airflow_cfm = air_distribution_airflows.max
+
+          fan_watts_per_cfm = get_fan_watts_per_cfm(hpxml_bldg)
+          adjusted_fan_watts_per_cfm = get_adjusted_fan_watts_per_cfm(baseline_max_airflow_cfm, upgrade_max_airflow_cfm, fan_watts_per_cfm)
+
+          measures['BuildResidentialHPXML'][0]['hvac_blower_fan_watts_per_cfm'] = adjusted_fan_watts_per_cfm
+        else # not ducted -> ducted; don't set autosizing limits or adjusted fan power
+          measures['BuildResidentialHPXML'][0].delete('heating_system_heating_autosizing_limit')
+          measures['BuildResidentialHPXML'][0].delete('cooling_system_cooling_autosizing_limit')
+          measures['BuildResidentialHPXML'][0].delete('heat_pump_heating_autosizing_limit')
+          measures['BuildResidentialHPXML'][0].delete('heat_pump_cooling_autosizing_limit')
+          measures['BuildResidentialHPXML'][0].delete('heat_pump_backup_heating_autosizing_limit')
+          measures['BuildResidentialHPXML'][0].delete('heating_system_2_heating_autosizing_limit')
+        end
+
+        if not apply_measures(hpxml_measures_dir, measures_hash, new_runner, model, true, 'OpenStudio::Measure::ModelMeasure', nil)
```

Comment:
Applying the BuildResidentialHPXML measure _again_ seems costly. Is there a way we could calculate/retrieve airflow rates for the upgraded building by just calling individual sizing method(s)?